src/common/snippets/src/lowered/pass/validate.cpp

```diff
@@ -0,0 +1,188 @@
+    if (is_type<snippets::op::RankNormalization>(first_consumer->get_node())) {
+        OPENVINO_ASSERT(consumer_inputs.size() == 1,
+                        "If there is RankNormalization after Parameter, it should be single consumer of the Parameter");
+        consumer_inputs = first_consumer->get_output_port_connector(0)->get_consumers();
+    }
+    std::set<std::vector<size_t>> layouts;
+    for (const auto& consumer_input : consumer_inputs) {
+        const auto& node = consumer_input.get_expr()->get_node();
+        if (const auto ma = ov::as_type_ptr<snippets::op::MemoryAccess>(node)) {
+            OPENVINO_ASSERT(ma->is_memory_access_input_port(consumer_input.get_index()),
+                            "Parameter expects MemoryAccess on output");
+            layouts.insert(consumer_input.get_descriptor_ptr()->get_layout());
+        } else if (ov::is_type<op::LoopEnd>(node)) {
+            continue;
+        } else {
+            OPENVINO_THROW("Parameter must be connected to MemoryAccess op or LoopEnd");
+        }
+    }
+    OPENVINO_ASSERT(layouts.size() == 1, "All consumers of Parameter must have the same layout");
+}
+
+void validate_result(const ExpressionPtr& expr, const LinearIR& linear_ir) {
+    OPENVINO_ASSERT(ov::is_type<ov::op::v0::Result>(expr->get_node()),
+                    "Result validation expects Result op");
+    const auto source = expr->get_input_port_connector(0)->get_source();
+    const auto ma = ov::as_type_ptr<snippets::op::MemoryAccess>(source.get_expr()->get_node());
+    OPENVINO_ASSERT(ma && ma->is_memory_access_output_port(source.get_index()),
+                    "Result expects MemoryAccess parent");
+}
+
+void validate_buffer(const ExpressionPtr& expr, const LinearIR& linear_ir) {
+    OPENVINO_ASSERT(ov::is_type<op::Buffer>(expr->get_node()),
+                    "Buffer validation expects Buffer op");
+    for (const auto& in : expr->get_input_port_connectors()) {
+        const auto& source = in->get_source();
+        const auto ma = ov::as_type_ptr<snippets::op::MemoryAccess>(source.get_expr()->get_node());
+        OPENVINO_ASSERT(ma && ma->is_memory_access_input_port(source.get_index()),
+                        "Buffer expects MemoryAccess parent");
+    }
+    for (const auto& out : expr->get_output_port_connectors()) {
+        const auto consumers = out->get_consumers();
+        for (const auto& consumer_input : consumers) {
+            const auto& node = consumer_input.get_expr()->get_node();
+            if (const auto ma = ov::as_type_ptr<snippets::op::MemoryAccess>(node)) {
+                OPENVINO_ASSERT(ma->is_memory_access_input_port(consumer_input.get_index()),
+                                "Buffer expects MemoryAccess on output");
+            } else if (ov::is_type<op::LoopEnd>(node)) {
+                continue;
+            } else {
+                OPENVINO_THROW("Parameter must be connected to MemoryAccess op or LoopEnd");
+            }
```

Comment:
Minor: we can combine it in a single OPENVINO_ASSERT

Reply:
Too, done, thanks